pkg/storageincentives/agent.go

```diff
@@ -0,0 +1,368 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package storageincentives
+
+import (
+	"context"
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/log"
+	"github.com/ethersphere/bee/pkg/postage"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+const loggerName = "incentives"
+
+type ChainBackend interface {
+	BlockNumber(context.Context) (uint64, error)
+}
+
+type Sampler interface {
+	ReserveSample(context.Context, []byte, uint8) ([]byte, error)
```

Comment:
I think it will be better to use the actual signature of the sampler. If not we would have to write some adapter later in the node init code.

Reply:
good call, this was written before your mege